Persist in-memory filesystem between tool uses

rtfeldman opened the pull request:
Release Notes:
- N/A

rtfeldman committed:
Persist in-memory filesystem between tool uses

maxdeviant commented on crates/assistant_tool/src/assistant_tool.rs:
```diff
@@ -31,6 +34,7 @@ pub trait Tool: 'static + Send + Sync {
     fn run(
         self: Arc<Self>,
         input: serde_json::Value,
+        file_changes: Arc<Mutex<HashMap<PathBuf, Vec<u8>>>>,
```
I don't think putting it on the `Tool` interface makes sense, as this seems quite specific to the scripting tool.

What if we put it in a `gpui::Global` instead? I suppose we would need to key it by something if we wanted it to be per-thread, so maybe pass the thread ID as an `Arc<str>` to `Tool::run`?

mgsloan committed:
Use a global for scripting tool in-memory fs

Suggested in https://github.com/zed-industries/zed/pull/26132#discussion_r1981670385

rtfeldman commented:
We decided earlier today to actually write directly to the filesystem after all, so closing this!
